src/base/torrentfileguard.h

```diff
@@ -29,10 +29,11 @@
 #ifndef TOFFENTFILEGURAD_H
 #define TOFFENTFILEGURAD_H
 
+#include <QObject>
 #include <QString>
-#include <QMetaType>
 
-class QMetaEnum;
+template <typename T> class CachedSettingValue;
```

Comment:
It might make sense to create a set of headers with forward declarations for our base units: `bittorrentfwd.h`, `baseutilsfwd.h`, `basefwd.h`. What do you think?